textdb/textdb-exp/src/main/java/edu/uci/ics/textdb/exp/sink/AbstractSink.java

```diff
@@ -17,13 +19,18 @@
 public abstract class AbstractSink implements ISink {
 
     private IOperator inputOperator;
+    private int cursor = CLOSED;
 
     /**
      * @about Opens the child operator.
      */
     @Override
     public void open() throws TextDBException {
+        if (cursor != CLOSED) {
```

Comment:
Per our discussion, I believe it's better to throw an exception to reveal the problem early.